docs/architecture/adr-059-evidence-composition-and-lifecycle.md

```diff
@@ -0,0 +1,299 @@
+
+  - If equivocation the validators that signed for the commits of both the trusted and signed header
+
+  - If lunatic the validators from the common val set that signed in the conflicting block
+
+  - If amnesia none (we can't know which validators are malicious). This also means that we don't
+  send amnesia evidence to the application
+
+- For each validator, check the look up table to make sure there already isn't evidence against this validator
+
+After verification we persist the evidence with the following key: `height/hash` to the pending evidence database in the evidence pool with the following format
+
+```go
+type EvidenceInfo struct {
+  ev Evidence
+  time time.Time
+  validators []Validator
+  totalVotingPower int64
+}
+```
+
+`time`, `validators` and `totalVotingPower` are need to form the ABCI Evidence that we will come to later.
+
+
+#### Broadcasting and receiving evidence
+
+The evidence pool also runs a reactor that broadcasts the newly validated
+evidence to all connected peers.
+
+Receiving evidence from other evidence reactors works in the same manner as receiving evidence from the consensus reactor or a light client.
+
+
+#### Proposing evidence on the block
+
+When it comes to prevoting and precomitting a proposal that contains evidence, the full node will once again
+call upon the evidence pool to verify the evidence using `CheckEvidence(ev []Evidence)`:
+
+This performs the following actions:
+
+1. Loops through all the evidence to check that nothing has been duplicated
+
+2. For each evidence, run `fastCheck(ev evidence)` which works similar to `Has` but instead for `LightClientAttackEvidence` if it has the
+same hash it then goes on to check that the validators it has are all signers in the commit of the conflicting header. If it doesn't pass fast check (because it hasn't seen the evidence before) then it will have to verify the evidence.
+
+3. runs `Verify(ev Evidence)` - Note: this also saves the evidence to the db as mentioned before.
+
+
+#### Updating application and pool
+
+The final part of the lifecycle is when the block is committed and the `BlockExecutor` then updates state. As part of this process, the `BlockExecutor` gets the evidence pool to create a simplified format for the evidence to be sent to the application. This happens in `ApplyBlock` where the executor calls `Update(Block, State) []abci.Evidence`.
+
+```go
+abciResponses.BeginBlock.ByzantineValidators = evpool.Update(block, state)
+```
+
+Here is the format of the evidence that the application will receive. As seen above, this is stored as an array within `BeginBlock`.
+The changes to the application are minimal (it is still formed one for each malicious validator) with the exception of using an enum instead of a string for the evidence type.
```

Comment:
```suggestion
As far as the application is concerned, the changes introduced by Tendermint Core 0.34 are minimal. Previous versions of Tendermint also had an `abci.Evidence` type, and 0.34 continues to form one `Evidence` object for each malicious validator. The only change is that 0.34 uses an enum instead of a string for the `Type`.
```

(You don't need to accept this change; it's just a suggestion, trying to make it as clear as possible.)